pkg/asset/machines/baremetal/hosts.go

```diff
@@ -172,3 +193,97 @@ func Hosts(config *types.InstallConfig, machines []machineapi.Machine, userDataS
 
 	return settings, nil
 }
+
+// ArbiterHosts returns the HostSettings with details of the hardware being
+// used to construct a cluster with an arbiter node.
+func ArbiterHosts(config *types.InstallConfig, machines []machineapi.Machine, userDataSecret string) (*HostSettings, error) {
+	settings := &HostSettings{}
+
+	if config.Platform.BareMetal == nil {
+		return nil, fmt.Errorf("no baremetal platform in configuration")
+	}
+
+	// If Arbiter is not enabled, nothing happens
+	if !config.IsArbiterEnabled() {
+		return nil, nil
+	}
+
+	numRequiredMasters := replicaCount(config.ControlPlane)
+	numMasters := 0
+
+	numRequiredArbiters := len(machines)
+	numArbiters := 0
+	for _, host := range config.Platform.BareMetal.Hosts {
+		// We make sure to keep an accurate count of known masters when explicitly set
+		if host.IsMaster() {
+			numMasters++
+		}
+
+		// If we know we're not an arbiter and the role isn't an empty value
+		// no further investigating is needed, skip creation.
+		if !host.IsArbiter() && host.Role != "" {
+			continue
+		}
+
+		// When it comes to hosts, the masters should have been generated first we skip those hosts
+		// and grab the hosts after to satisfy the number of required arbiters, and the host isn't
+		// explicitly set as an arbiter, this allows for mixed role types in the list.
+		if numMasters < numRequiredMasters && !host.IsArbiter() {
```

Comment:
nit: `&& !host.IsArbiter()` is redundant here; we would have already continued on line 225 if `host.IsArbiter()` were true.

Reply:
This unfortunately is needed, but it's not very clear as to the intent here. I updated the logic to be what we're really checking for here, which is the situation where the `host.Role != ""` to account for empty roles correctly. I updated and hopefully added a clearer comment.